syncer/sharding_group.go

```diff
@@ -307,37 +307,46 @@ func (sg *ShardingGroup) UnresolvedTables() [][]string {
 	return tables
 }
 
-// FirstPosUnresolved returns the first DDL pos if un-resolved, else nil
-func (sg *ShardingGroup) FirstPosUnresolved() *mysql.Position {
+// FirstLocationUnresolved returns the first DDL pos if un-resolved, else nil
+func (sg *ShardingGroup) FirstLocationUnresolved() *binlog.Location {
 	sg.RLock()
 	defer sg.RUnlock()
-	if sg.remain < len(sg.sources) && sg.firstPos != nil {
+	if sg.remain < len(sg.sources) && sg.firstLocation != nil {
 		// create a new pos to return
-		return &mysql.Position{
-			Name: sg.firstPos.Name,
-			Pos:  sg.firstPos.Pos,
+		return &binlog.Location{
+			Position: mysql.Position{
+				Name: sg.firstLocation.Position.Name,
+				Pos:  sg.firstLocation.Position.Pos,
+			},
+			GTIDSet: sg.firstLocation.GTIDSet,
```

Comment:
yes, I add `Clone` in 6d3ab39052ef4bca5acb76d12b34566125b56397, and seems that don't need to judge whether it is nil, because we will make sure the `GTIDSet` from meta and checkpoint is not nil.